src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1124,20 +1124,97 @@ protected function xmlToProps($xml)
     protected function xmlToColumns($xml, $propertyNames)
     {
         $data = new stdClass();
-        $dom = new DOMDocument('1.0');
-        $dom->loadXML($xml);
-        $dom->formatOutput = true;
-        $xpath = new DOMXPath($dom);
 
-        foreach ($propertyNames as $propertyName) {
-            $els = $xpath->query(sprintf('.//sv:property[@sv:name="%s"]', $propertyName));
-            if ($els->length === 0) {
-                continue;
+        $lastPropertyName = null;
+        $lastPropertyType = null;
+        $lastPropertyMultiValued = null;
+        $currentTag = null;
+        $currentValues = [];
+        $currentValue = null;
+
+        $parser = xml_parser_create();
+        xml_set_element_handler(
+            $parser,
+            function ($parser, $name, $attrs) use ($propertyNames, &$lastPropertyName, &$lastPropertyType, &$lastPropertyMultiValued, &$currentTag) {
+                $currentTag = $name;
+
+                if ($name === 'SV:PROPERTY') {
+                    if (!\in_array($attrs['SV:NAME'], $propertyNames)) {
+                        return;
+                    }
+                    $lastPropertyName = $attrs['SV:NAME'];
+                    $lastPropertyType = PropertyType::valueFromName($attrs['SV:TYPE']);
+                    $lastPropertyMultiValued = $attrs['SV:MULTI-VALUED'];
+                }
+            },
+            function ($parser, $name) use (&$data, &$lastPropertyName, &$lastPropertyType, &$lastPropertyMultiValued, &$currentTag, &$currentValues, &$currentValue) {
+                if ($name === 'SV:PROPERTY' && $lastPropertyName) {
+                    switch ($lastPropertyType) {
+                        case PropertyType::BINARY:
+                            $data->{':' . $lastPropertyName} = $lastPropertyMultiValued ? $currentValues : $currentValues[0];
+                            break;
+                        default:
+                            $data->{$lastPropertyName} = $lastPropertyMultiValued ? $currentValues : $currentValues[0];
+                            $data->{':' . $lastPropertyName} = $lastPropertyType;
+                            break;
+                    }
+
+                    $currentValues = [];
+                    $lastPropertyName = null;
+                    $lastPropertyType = null;
+                    $lastPropertyMultiValued = null;
+                } elseif ($name === 'SV:VALUE' && $lastPropertyName) {
```

Comment:
can we return abvoe, instead of the elseif?

Reply:
sure will do this :)